### PR TITLE
[Aptos Framework] Table with aggregator length and optional aggregator

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

This PR adds 2 new Move modules:

1. Table implementation which uses aggregator to avoid the bottleneck on updating the table length.
2. Optional aggregator struct which acts as a normal integer or an aggregator based on the
     initialization.

Also, all experimental sources are within `stdlib` now to avoid cyclic dependencies.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2706)

### CODE REVIEW

**Comment by zekun000** [comment]
do you want to keep both the optional aggregator as well as the table with aggregator?

**Comment by georgemitenkov** [comment]
@zekun000 both can be useful I guess. Added it here mostly to see what others think. cc @gelash @davidiw @movekevin ?

File: aptos-move/framework/aptos-stdlib/sources/experimental/optional_aggregator.move [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    fun new_integer(limit: u128): Integer {
+        Integer {
+            value: 0,
+            limit,
+        }
+    }
+
+    fun add_integer(base: &mut Integer, value: u128) {
+        assert!(
+            base.limit >= base.value && value <= (base.limit - base.value),
```
Comment (junkil-park) [inline aptos-move/framework/aptos-stdlib/sources/experimental/optional_aggregator.move:32]: It seems that `base.limit >= base.value` is always true, so unnecessary to check. No function including `new_integer` and `add_integer` can break this property (or invariant).

**Review by wrwg — APPROVED** [review]
Let's get it in. After all it is clearly marked as experimental ;-)

**Review by gelash — APPROVED** [review]

**Pull request merged**